exercises/rational-numbers/canonical-data.json

```diff
@@ -0,0 +1,356 @@
+              "property": "mul",
+              "input": {
+                "r1": [-1, 2],
+                "r2": [2, 3]
+              },
+              "expected": [-1, 3]
+            },
+            {
+              "description": "Multiply two negative rational numbers",
+              "property": "mul",
+              "input": {
+                "r1": [-1, 2],
+                "r2": [-2, 3]
+              },
+              "expected": [1, 3]
+            },
+            {
+              "description": "Multiply a rational number by its reciprocal",
+              "property": "mul",
+              "input": {
+                "r1": [1, 2],
+                "r2": [2, 1]
+              },
+              "expected": [1, 1]
+            },
+            {
+              "description": "Multiply a rational number by 1",
+              "property": "mul",
+              "input": {
+                "r1": [1, 2],
+                "r2": [1, 1]
+              },
+              "expected": [1, 2]
+            },
+            {
+              "description": "Multiply a rational number by 0",
+              "property": "mul",
+              "input": {
+                "r1": [1, 2],
+                "r2": [0, 1]
+              },
+              "expected": [0, 1]
+            }
+          ]
+        },
+        {
+          "description": "Division",
+          "cases": [
+            {
+              "description": "Divide two positive rational numbers",
+              "property": "div",
+              "input": {
+                "r1": [1, 2],
+                "r2": [2, 3]
+              },
+              "expected": [3, 4]
+            },
+            {
+              "description": "Divide a positive rational number by a negative rational number",
+              "property": "div",
+              "input": {
+                "r1": [1, 2],
+                "r2": [-2, 3]
+              },
+              "expected": [-3, 4]
+            },
+            {
+              "description": "Divide two negative rational numbers",
+              "property": "div",
+              "input": {
+                "r1": [-1, 2],
+                "r2": [-2, 3]
+              },
+              "expected": [3, 4]
+            },
+            {
+              "description": "Divide a rational number by 1",
+              "property": "div",
+              "input": {
+                "r1": [1, 2],
+                "r2": [1, 1]
+              },
+              "expected": [1, 2]
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "description": "Absolute value",
+      "cases": [
+        {
+          "description": "Absolute value of a positive rational number",
+          "property": "abs",
+          "input": [1, 2],
+          "expected": [1, 2]
+        },
+        {
+          "description": "Absolute value of a negative rational number",
+          "property": "abs",
+          "input": [-1, 2],
```

Comment:
> let's make `input` an object with named keys/values, instead of an array like it is on this line
> 
> the reason to do this would be that:
> 
> * it is consistent with the rest of the cases
> * it is compliant with https://github.com/exercism/problem-specifications/issues/996#issuecomment-342740372

same